python/orca/src/bigdl/orca/data/shard.py

```diff
@@ -377,7 +365,8 @@ def merge_list_of_dict(iter):
                 repartitioned_shard = SparkXShards(self.rdd.repartition(num_partitions))
         else:
             repartitioned_shard = SparkXShards(self.rdd.repartition(num_partitions))
-        self._uncache()
+        if not transient:
+            self._uncache()
```

Comment:
Instead of using `transient` here, you may want to explicitly call `SparkXShards.cache` in the estimator